core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -263,6 +263,9 @@ static void init(String[] args) throws Throwable {
         INSTANCE = new Bootstrap();
 
         boolean foreground = !"false".equals(System.getProperty("es.foreground", System.getProperty("es-foreground")));
+        // Clear these properties because we are done with them. No need to let them leak into the Settings.
+        System.clearProperty("es.foreground");
```

Comment:
This only "works" because we have not yet installed security manager at this point. I don't think we should be modifying system properties. What purpose does this serve?